--- com/moonsworth/lunar/v1_8/mods/replaymod/recording/handler/RecordingEventHandler.java
+++ com/moonsworth/lunar/v1_8/mods/replaymod/recording/handler/RecordingEventHandler.java
@@ -1,5 +1,4 @@
 package com.moonsworth.lunar.v1_8.mods.replaymod.recording.handler;
 
-import com.replaymod.recording.mixin.IntegratedServerAccessor;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.entity.EntityPlayerSP;
@@ -79,6 +78,6 @@
 
     @Listen
-    public void onPlayerTick(EventTick.ClientTickEvent e) {
-        if (e.phase != EventTick.Phase.START || mc.thePlayer == null)
+    public void onPlayerTick(EventTick e) {
+        if (mc.thePlayer == null)
             return;
         EntityPlayerSP player = mc.thePlayer;
@@ -214,5 +213,5 @@
     public void onPickupItem(ItemPickupEvent event) {
         try {
-            packetListener.save(new S0DPacketCollectItem(event.pickedUp.getEntityId(), event.player.getEntityId()));
+            packetListener.save(new S0DPacketCollectItem(event.getEntityItem().bridge$getEntityId(), event.getPlayer().bridge$getEntityId()));
         } catch (Exception e) {
             e.printStackTrace();
@@ -223,9 +222,9 @@
     public void onSleep(EventPlayerSleepInBed event) {
         try {
-            if (event.entityPlayer != mc.thePlayer) {
+            if (!((EntityPlayer)event.getPlayer()).equals(mc.thePlayer)) {
                 return;
             }
 
-            packetListener.save(new S0APacketUseBed(event.entityPlayer, event.pos));
+            packetListener.save(new S0APacketUseBed((EntityPlayer) event.getPlayer(), (BlockPos) event.getPos()));
 
             wasSleeping = true;
@@ -236,31 +235,4 @@
     }
 
-    /* FIXME event not (yet?) on 1.13
-    @SubscribeEvent
-    public void enterMinecart(MinecartInteractEvent event) {
-        try {
-            //#if MC>=10904
-            //$$ if(event.getEntity() != mc.player) {
-            //$$     return;
-            //$$ }
-            //$$
-            //$$ packetListener.save(new SPacketEntityAttach(event.getPlayer(), event.getMinecart()));
-            //$$
-            //$$ lastRiding = event.getMinecart().getEntityId();
-            //#else
-            if(event.entity != mc.thePlayer) {
-                return;
-            }
-    
-            packetListener.save(new S1BPacketEntityAttach(0, event.player, event.minecart));
-    
-            lastRiding = event.minecart.getEntityId();
-            //#endif
-        } catch(Exception e) {
-            e.printStackTrace();
-        }
-    }
-    */
-
     public void onBlockBreakAnim(int breakerId, BlockPos pos, int progress) {
         EntityPlayer thePlayer = mc.thePlayer;
@@ -271,10 +243,8 @@
 
     @Listen
-    public void checkForGamePaused(EventTick.RenderTickEvent event) {
-        if (event.phase != EventTick.Phase.START)
-            return;
+    public void checkForGamePaused(EventTick event) {
         if (mc.isSingleplayer()) {
             IntegratedServer server = mc.getIntegratedServer();
-            if (server != null && ((IntegratedServerAccessor) server).isGamePaused()) {
+            if (server != null && server.isGamePaused) {
                 packetListener.setServerWasPaused();
             }
