--- com/moonsworth/lunar/v1_8/mods/replaymod/replay/ReplaySender.java
+++ com/moonsworth/lunar/v1_8/mods/replaymod/replay/ReplaySender.java
@@ -1,6 +1,4 @@
 package com.moonsworth.lunar.v1_8.mods.replaymod.replay;
 
-import com.replaymod.core.mixin.MinecraftAccessor;
-import com.replaymod.core.mixin.TimerAccessor;
 import net.minecraft.client.Minecraft;
 
@@ -16,6 +14,5 @@
     public default boolean paused() {
         Minecraft mc = getMinecraft();
-        TimerAccessor timer = (TimerAccessor) ((MinecraftAccessor) mc).getTimer();
-        return timer.getTimerSpeed() == 0;
+        return mc.timer.timerSpeed == 0;
     }
 
