--- com/moonsworth/lunar/v1_8/mods/replaymod/mixin/extras/playeroverview/RenderMixin_ReplayMod.java
+++ com/moonsworth/lunar/v1_8/mods/replaymod/mixin/extras/playeroverview/RenderMixin_ReplayMod.java
@@ -8,5 +8,5 @@
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.Coerce;
+import net.minecraft.client.renderer.culling.ICamera;
 import org.spongepowered.asm.mixin.injection.Inject;
 
@@ -31,5 +31,5 @@
 public abstract class RenderMixin_ReplayMod {
     @Inject(method = "shouldRender", at = @At("HEAD"), cancellable = true)
-    public void replayModExtras_isPlayerHidden(Entity entity, @Coerce Object camera, double camX, double camY,
+    public void replayModExtras_isPlayerHidden(Entity entity, ICamera camera, double camX, double camY,
             double camZ, CallbackInfoReturnable<Boolean> ci) {
         ReplayModExtras.instance.get(PlayerOverview.class).ifPresent(playerOverview -> {
