--- com/moonsworth/lunar/v1_8/mods/replaymod/replay/gui/screen/GuiModCompatWarning.java
+++ com/moonsworth/lunar/v1_8/mods/replaymod/replay/gui/screen/GuiModCompatWarning.java
@@ -9,5 +9,4 @@
 import com.moonsworth.lunar.v1_8.mods.replaymod.minecraft.gui.layout.HorizontalLayout;
 import com.moonsworth.lunar.v1_8.mods.replaymod.minecraft.gui.layout.VerticalLayout;
-import com.replaymod.core.utils.ModCompat;
 import com.replaymod.replaystudio.data.ModInfo;
 import com.replaymod.replaystudio.util.I18n;
@@ -35,50 +34,50 @@
     }
 
-    public GuiModCompatWarning(ModCompat.ModInfoDifference difference) {
-        VerticalLayout.Data data = new VerticalLayout.Data(0.5);
-        GuiPanel content = this.content.getListPanel();
-        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message1"));
-        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message2"));
-        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message3"));
+//    public GuiModCompatWarning(ModCompat.ModInfoDifference difference) {
+//        VerticalLayout.Data data = new VerticalLayout.Data(0.5);
+//        GuiPanel content = this.content.getListPanel();
+//        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message1"));
+//        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message2"));
+//        content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.message3"));
+//
+//        if (!difference.getMissing().isEmpty()) {
+//            content.addElements(data, new GuiLabel());
+//            content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.missing"));
+//            for (ModInfo modInfo : difference.getMissing()) {
+//                content.addElements(data,
+//                        new GuiPanel().setLayout(new VerticalLayout().setSpacing(3)).addElements(null,
+//                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
+//                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.name"), null)
+//                                        .with(new GuiLabel().setText(modInfo.getName()), null).build(),
+//                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
+//                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.id"), null)
+//                                        .with(new GuiLabel().setText(modInfo.getId()), null).build(),
+//                                new GuiLabel().setText(I18n.format("replaymod.gui.modwarning.version.expected") + ": "
+//                                        + modInfo.getVersion())));
+//            }
+//        }
+//
+//        if (!difference.getDiffering().isEmpty()) {
+//            content.addElements(data, new GuiLabel());
+//            content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.version"));
+//            for (Map.Entry<ModInfo, String> entry : difference.getDiffering().entrySet()) {
+//                content.addElements(data,
+//                        new GuiPanel().setLayout(new VerticalLayout().setSpacing(3)).addElements(null,
+//                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
+//                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.name"), null)
+//                                        .with(new GuiLabel().setText(entry.getKey().getName()), null).build(),
+//                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
+//                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.id"), null)
+//                                        .with(new GuiLabel().setText(entry.getKey().getId()), null).build(),
+//                                new GuiLabel().setText(I18n.format("replaymod.gui.modwarning.version.expected") + ": "
+//                                        + entry.getKey().getVersion() + ", "
+//                                        + I18n.format("replaymod.gui.modwarning.version.found") + ": "
+//                                        + entry.getValue())));
+//            }
+//        }
+//
+//        cancelButton.onClick(() -> getMinecraft().displayGuiScreen(null));
+//    }
 
-        if (!difference.getMissing().isEmpty()) {
-            content.addElements(data, new GuiLabel());
-            content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.missing"));
-            for (ModInfo modInfo : difference.getMissing()) {
-                content.addElements(data,
-                        new GuiPanel().setLayout(new VerticalLayout().setSpacing(3)).addElements(null,
-                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
-                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.name"), null)
-                                        .with(new GuiLabel().setText(modInfo.getName()), null).build(),
-                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
-                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.id"), null)
-                                        .with(new GuiLabel().setText(modInfo.getId()), null).build(),
-                                new GuiLabel().setText(I18n.format("replaymod.gui.modwarning.version.expected") + ": "
-                                        + modInfo.getVersion())));
-            }
-        }
-
-        if (!difference.getDiffering().isEmpty()) {
-            content.addElements(data, new GuiLabel());
-            content.addElements(data, new GuiLabel().setI18nText("replaymod.gui.modwarning.version"));
-            for (Map.Entry<ModInfo, String> entry : difference.getDiffering().entrySet()) {
-                content.addElements(data,
-                        new GuiPanel().setLayout(new VerticalLayout().setSpacing(3)).addElements(null,
-                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
-                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.name"), null)
-                                        .with(new GuiLabel().setText(entry.getKey().getName()), null).build(),
-                                GuiPanel.builder().layout(new HorizontalLayout().setSpacing(3))
-                                        .with(new GuiLabel().setI18nText("replaymod.gui.modwarning.id"), null)
-                                        .with(new GuiLabel().setText(entry.getKey().getId()), null).build(),
-                                new GuiLabel().setText(I18n.format("replaymod.gui.modwarning.version.expected") + ": "
-                                        + entry.getKey().getVersion() + ", "
-                                        + I18n.format("replaymod.gui.modwarning.version.found") + ": "
-                                        + entry.getValue())));
-            }
-        }
-
-        cancelButton.onClick(() -> getMinecraft().displayGuiScreen(null));
-    }
-
     @Override
     protected GuiModCompatWarning getThis() {
