--- com/moonsworth/lunar/v1_8/mods/replaymod/extras/Setting.java
+++ com/moonsworth/lunar/v1_8/mods/replaymod/extras/Setting.java
@@ -10,10 +10,10 @@
     public static final SettingsRegistry.SettingKey<Boolean> SKIP_POST_SCREENSHOT_GUI = new SettingsRegistry.SettingKeys<>(
             "advanced", "skipPostScreenshotGui", null, false);
-    public static final SettingsRegistry.MultipleChoiceSettingKeys<String> FULL_BRIGHTNESS = new SettingsRegistry.MultipleChoiceSettingKeys<>(
-            "advanced", "fullBrightness", "replaymod.gui.settings.fullbrightness",
-            FullBrightness.Type.Gamma.toString());
+//    public static final SettingsRegistry.MultipleChoiceSettingKeys<String> FULL_BRIGHTNESS = new SettingsRegistry.MultipleChoiceSettingKeys<>(
+//            "advanced", "fullBrightness", "replaymod.gui.settings.fullbrightness",
+//            FullBrightness.Type.Gamma.toString());
     static {
-        FULL_BRIGHTNESS.setChoices(
-                Arrays.stream(FullBrightness.Type.values()).map(Object::toString).collect(Collectors.toList()));
+//        FULL_BRIGHTNESS.setChoices(
+//                Arrays.stream(FullBrightness.Type.values()).map(Object::toString).collect(Collectors.toList()));
     }
 }
